backends/graphics/atari/atari-graphics.cpp

```diff
@@ -0,0 +1,998 @@
+		: (byte*)allocFast(width * height * format.bytesPerPixel + 15);
+
+	if (!buf)
+		return false;
+
+	byte *bufAligned = (byte*)((((uintptr)buf + 15) | mask) & 0xfffffff0);
+	memset(bufAligned, 0, width * height * format.bytesPerPixel);
+
+	surface.init(width, height, width * format.bytesPerPixel, bufAligned, format);
+	return true;
+}
+
+void AtariGraphicsManager::setVidelResolution() const
+{
+	if (_vgaMonitor) {
+		// TODO: aspect ratio correction
+		// TODO: supervidel 320x240...
+		if (_screenSurface.w == 320) {
+			if (_screenSurface.h == 200)
+				asm_screen_set_scp_res(scp_320x200x8_vga);
+			else
+				asm_screen_set_scp_res(scp_320x240x8_vga);
+		} else {
+			if (_screenSurface.h == 400)
+				asm_screen_set_scp_res(scp_640x400x8_vga);
+			else
+				asm_screen_set_scp_res(scp_640x480x8_vga);
+		}
+	} else {
+		if (_screenSurface.w == 320) {
+			if (_screenSurface.h == 240)
+				asm_screen_set_scp_res(scp_320x240x8_rgb);
+			else if (_screenSurface.h == 200 && _aspectRatioCorrection)
+				asm_screen_set_scp_res(scp_320x200x8_rgb60);
+			else
+				asm_screen_set_scp_res(scp_320x200x8_rgb);
+		} else {
+			if (_screenSurface.h == 480)
+				asm_screen_set_scp_res(scp_640x480x8_rgb);
+			else if (_screenSurface.h == 400 && _aspectRatioCorrection)
+				asm_screen_set_scp_res(scp_640x400x8_rgb60);
+			else
+				asm_screen_set_scp_res(scp_640x400x8_rgb);
+		}
+	}
+}
+
+void AtariGraphicsManager::waitForVbl() const
+{
+	extern volatile uint32 vbl_counter;
+	uint32 counter = vbl_counter;
+
+	while (counter == vbl_counter);
+}
+
+bool AtariGraphicsManager::updateOverlay() {
+	bool updated = false;
+	bool drawCursor = _cursor.isModified();
+
+	while (!_modifiedOverlayRects.empty()) {
+		const Common::Rect &rect = _modifiedOverlayRects.back();
+
+		if (!drawCursor && !_cursor.outOfScreen && _cursor.visible)
+			drawCursor = rect.intersects(_cursor.dstRect);
+
+		_screenOverlaySurface.copyRectToSurface(_overlaySurface, rect.left, rect.top, rect);
+
+		_modifiedOverlayRects.pop_back();
+
+		updated = true;
+	}
+
+	if (_cursor.outOfScreen)
+		return updated;
+
+	if ((_cursor.positionChanged || !_cursor.visible) && !_oldCursorRect.isEmpty()) {
+		_screenOverlaySurface.copyRectToSurface(_overlaySurface, _oldCursorRect.left, _oldCursorRect.top, _oldCursorRect);
+		_oldCursorRect = Common::Rect();
+
+		updated = true;
+	}
+
+	if (drawCursor && _cursor.visible) {
+		//debug("Redraw cursor (overlay): %d %d %d %d", _cursor.dstRect.left, _cursor.dstRect.top, _cursor.dstRect.width(), _cursor.dstRect.height());
+
+		copySurface8ToSurface16WithKey(
+			_cursor.surface,
+			_cursor.palette,
+			_screenOverlaySurface,
+			_cursor.dstRect.left, _cursor.dstRect.top,
+			_cursor.srcRect,
+			_cursor.keycolor);
+
+		_cursor.positionChanged = _cursor.surfaceChanged = false;
+		_oldCursorRect = _cursor.dstRect;
+
+		updated = true;
+	}
+
+	return updated;
+}
+
+bool AtariGraphicsManager::updateDirectBuffer()
+{
```

Comment:
Please check all methods for the code formatting. I'll skip further cases like this.